tcpclient/connectionsManagement.go

```diff
@@ -1,30 +1,46 @@
 package tcpclient
 
 import (
-	"sync"
+	"bufio"
 	"fmt"
+	"io"
 	"net"
-	"bufio"
 	"strconv"
+	"sync"
 )
 
-func TcpConnect(id int, host string, port int, wg *sync.WaitGroup) error {
-	fmt.Println("\t runner " + strconv.Itoa(id) + " is initiating a connection")
+func reportConnectionStatus(debugOut io.Writer, statusChannel chan<- Connection, connectionDescription Connection)  {
+	statusChannel <- connectionDescription
+	fmt.Fprintln(debugOut, "\t", connectionDescription)
+}
+
+func TcpConnect(id int, host string, port int, wg *sync.WaitGroup, debugOut io.Writer, statusChannel chan<- Connection) error {
+	connectionDescription := Connection{
+		Id: id,
+		status: ConnectionDialing,
+	}
+	reportConnectionStatus(debugOut, statusChannel, connectionDescription)
 	conn, err := net.Dial("tcp", host+":"+strconv.Itoa(port))
 	if err != nil {
-		fmt.Println(err)
+		connectionDescription.status = ConnectionError
+		reportConnectionStatus(debugOut, statusChannel, connectionDescription)
+		fmt.Fprintln(debugOut, "Connection", id, "was unable to open the connection. Error:")
```

Comment:
why not summarize both `fmt.Fprintln` in one?

Reply:
Fprintln adds an EOL at the end of each call. And I think keeping the error in a different line will provide better output, but we can merge into the same line, or change Fprintln and take control of the \n... But it does not hurt too much, right?

Reply:
👍